CourseSketchProjects/IdentityServer/src/main/java/coursesketch/database/identity/IdentityManager.java

```diff
@@ -0,0 +1,549 @@
+                    .getBytes(StandardCharsets.UTF_8));
+        } catch (NoSuchAlgorithmException e) {
+            throw new AuthenticationException(e);
+        }
+
+        if (hash == null) {
+            throw new AuthenticationException("Unable to create authentication hash for group " + groupId, AuthenticationException.OTHER);
+        }
+
+        // final BasicDBObject newIdentity = new BasicDBObject(userId, hash);
+        final String list = isUser ? DatabaseStringConstants.USER_LIST : DatabaseStringConstants.NON_USER_LIST;
+        database.getCollection(DatabaseStringConstants.USER_GROUP_COLLECTION).update(
+                group,
+                new BasicDBObject(DatabaseStringConstants.SET_COMMAND, new BasicDBObject(list + "." + userId, hash)));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void registerUserInItem(final String userId, final String authId, final String itemId, final School.ItemType itemType,
+            final Authenticator authChecker) throws AuthenticationException, DatabaseAccessException {
+        if (!School.ItemType.COURSE.equals(itemType) && !School.ItemType.BANK_PROBLEM.equals(itemType)) {
+            throw new AuthenticationException("Can only register users in a course or a bank problem", AuthenticationException.OTHER);
+        }
+
+        final DBCollection collection = database.getCollection(getCollectionFromType(itemType));
+        final DBObject result = collection.findOne(new ObjectId(itemId),
+                new BasicDBObject(DatabaseStringConstants.USER_LIST, true));
+
+        if (result == null) {
+            throw new DatabaseAccessException("The item with the id " + itemId + " Was not found in the database");
+        }
+
+        final AuthenticationResponder responder = authChecker.checkAuthentication(itemType, itemId, authId, 0,
+                Authentication.AuthType.newBuilder().setCheckingAdmin(true).build());
+        if (!responder.hasStudentPermission()) {
+            throw new AuthenticationException("User does not have permission to insert new items for id: " + itemId,
+                    AuthenticationException.INVALID_PERMISSION);
+        }
+
+        final List<String> userGroups = (List<String>) result.get(DatabaseStringConstants.USER_LIST);
+        insertUserIntoGroup(userId, userGroups.get(0), !responder.hasPeerTeacherPermission());
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<String, String> createNewUser(final String userName) throws AuthenticationException, DatabaseAccessException {
+        final ObjectId userId = new ObjectId();
+        final String userPassword = AbstractServerWebSocketHandler.Encoder.nextID().toString();
+        String hashPassword;
+        try {
+            hashPassword = HashManager.createHash(userPassword);
+        } catch (NoSuchAlgorithmException e) {
+            LOG.error("Algorithm could not be found", e);
```

Comment:
Do you need this log, cause you don't log any where else before this when you catch a NoSuchAlgorithmException